packages/kbn-es-query/src/utils.ts

```diff
@@ -7,10 +7,31 @@
  */
 
 import moment from 'moment-timezone';
+import { DataViewFieldBase, IFieldSubTypeNested, IFieldSubTypeMulti } from './es_query';
 
 /** @internal */
 export function getTimeZoneFromSettings(dateFormatTZ: string) {
   const detectedTimezone = moment.tz.guess();
 
   return dateFormatTZ === 'Browser' ? detectedTimezone : dateFormatTZ;
 }
+
+type HasSubtype = Pick<DataViewFieldBase, 'subType'>;
+
+export function isDataViewFieldSubtypeNested(field: HasSubtype) {
```

Comment:
Perhaps you know how to do this but I don't - how to check if a given object is a specific typescript type. As best I can tell, this isn't possible. You can check if something is a specific javascript class but thats not what we have here and it would be a lot of work to implement.